### PR TITLE
Bazel: offload BUILD files

### PR DESCRIPTION
**Pull request by mrtnzlml**
This commit introduces some basic BUILD files to offload the main Bazel branch. It runs Bazel via Bazelisk on CI and currently executes only two jobs: Buldifier to lint BUILD and WORKSPACE files and `bazel build`.

It should not be used yet so there is no documentation for it yet. I will keep offloading things in a non-breaking fashion in other commits.

### CODE REVIEW

File: WORKSPACE.bazel [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    sha256 = "b725e6497741d7fc2d55fcc29a276627d10e43fa5d0bb692692890ae30d98d00",
+    urls = [
+        "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.24.3/rules_go-v0.24.3.tar.gz",
+        "https://github.com/bazelbuild/rules_go/releases/download/v0.24.3/rules_go-v0.24.3.tar.gz",
+    ],
+)
+
+load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")
+
+go_rules_dependencies()
```
Comment (tbergq) [inline WORKSPACE.bazel:54]: What are we using go_rules for?

Comment (mrtnzlml) [reply WORKSPACE.bazel:54]: Buldifier is written in Go and it requires to be set like this, see: https://github.com/bazelbuild/buildtools/blob/e6efbf6df90bec363c3cbd564b72be6c8a309f14/buildifier/README.md

File: bazel/macros/build_yarn_workspace.bzl [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            "src/**/*.test.js",
+        ],
+        data = [
+            ":package.json",
+            ":src",
+            "//:babel.config.js",
+            "//:package.json",
+            "@npm//@adeira/babel-preset-adeira",
+        ],
+    )
```
Comment (tbergq) [inline bazel/macros/build_yarn_workspace.bzl:29]: We are not building flow and es-modules outputs?

**Review by tbergq — APPROVED** [review]
Looks like a good start 👍

File: bazel/macros/build_yarn_workspace.bzl (hunk 2 above)
Comment (mrtnzlml) [reply bazel/macros/build_yarn_workspace.bzl:29]: Not yet. But I want to figure out how to do it before considering this to be stable. It should not be part of this build step I think. This build is mainly for other parts of the monorepo to be able to use the built versions of our packages.

Comment (tbergq) [reply bazel/macros/build_yarn_workspace.bzl:29]: Sure, makes sense, one way is to build `lib` folder and `es` folder, and have main and module point to the respective folders, but I don't know how to put in the flow files yet 🤕

Comment (mrtnzlml) [reply bazel/macros/build_yarn_workspace.bzl:29]: Stay tuned, we are going to figure it out. I also don't know how to do it quite yet.

**Pull request merged**